kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerOffsetsDaoImpl.java

```diff
@@ -0,0 +1,165 @@
+      return new AutoValue_ConsumerOffsetsDaoImpl_MemberId.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      abstract Builder setConsumerId(String consumerId);
+
+      abstract Builder setClientId(String clientId);
+
+      abstract Builder setInstanceId(Optional<String> instanceId);
+
+      abstract MemberId build();
+    }
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, MemberId>> getMemberIds(
+      ConsumerGroupDescription cgDesc
+  ) {
+    Map<TopicPartition, MemberId> tpMemberIds = new HashMap<>();
+    for (MemberDescription memberDesc : cgDesc.members()) {
+      for (TopicPartition tp : memberDesc.assignment().topicPartitions()) {
+        MemberId memberId =
+            MemberId.builder()
+                .setConsumerId(memberDesc.consumerId())
+                .setClientId(memberDesc.clientId())
+                .setInstanceId(memberDesc.groupInstanceId())
+                .build();
+        tpMemberIds.put(tp, memberId);
+      }
+    }
+    return CompletableFuture.completedFuture(tpMemberIds);
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> getCurrentOffsets(
+      String consumerGroupId
+  ) {
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.listConsumerGroupOffsets(
+            consumerGroupId, new ListConsumerGroupOffsetsOptions())
+            .partitionsToOffsetAndMetadata());
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, ListOffsetsResultInfo>> getLatestOffsets(
+      IsolationLevel isolationLevel,
+      Map<TopicPartition, OffsetAndMetadata> currentOffsets
+  ) {
+    Map<TopicPartition, OffsetSpec> latestOffsetSpecs =
+        currentOffsets.keySet()
+            .stream()
+            .collect(
+                Collectors.toMap(Function.identity(), tp -> OffsetSpec.latest()));
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.listOffsets(
+            latestOffsetSpecs,
+            new ListOffsetsOptions(isolationLevel)).all());
+  }
+
+  @Override
+  public CompletableFuture<Optional<ConsumerGroupDescription>> getConsumerGroupDescription(
+      String consumerGroupId) {
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.describeConsumerGroups(Collections.singletonList(consumerGroupId)).all())
+        .thenApply(
+            descriptions ->
+                descriptions.values()
+                    .stream()
+                    .filter(
+                        // When describing a consumer-group that does not exist, AdminClient returns
+                        // a dummy consumer-group with simple=true and state=DEAD.
+                        // TODO: Investigate a better way of detecting non-existent consumer-group.
+                        description -> !description.isSimpleConsumerGroup()
+                            || description.state() != ConsumerGroupState.DEAD)
+                    .findAny());
+  }
```

Comment:
Why do you need this method at all? Why don't you inject ConsumerGroupManager and use getConsumerGroup instead? Is there something in ConsumerGroupDescription which is not reflected in ConsumerGroup? If that is the case, would not it be better to augment ConsumerGroup instead, and ConsumerGroupManager?